src/en/developer-layers.md

```diff
@@ -0,0 +1,153 @@
+
+ Title: Layers for Charm Authoring
+
+# Why build from layers
+
+When creating a charm, you always have the option of doing it by
+hand writing.  That is, you can create each hook, implement the side of the
+interface you need for each relation your charm requires or provides, manage the
+dependencies, such as [charm-helpers][https://pythonhosted.org/charmhelpers/],
+that your charm uses, et cetera. However, what you really want to do is focus on
+*your* charm.  So, why not leverage the work of others and keep your charm code
+as minimal and tightly focused as possible?
+
+Enter the concept of building charms from layers.  Layers let you build on the
+work of other charmers, whether that work is in the form of other charms that
+you can extend and modify, interfaces that are already built for you and know
+how to communicate with a remote service and let you know when that service is
+ready and what it provides for you, or partial base layers that make managing
+dependencies much easier.  And it does this in a consistent, repeatable, and
+audit-able way.
+
+# What are layers?
+
+Layers are encapsulated charm code which lend themselves to being re-used across
+charms. They come in distinct flavors:
+
+- Base, or runtime layers Interface layers Charm layers
```

Comment:
layers layers layers!

Are you missing commas in this sentence?  "Base, runtime, interface, or charm layers."